lib/meilisearch-rails.rb

```diff
@@ -306,7 +306,7 @@ def settings(*args)
         SafeIndex.log_or_throw(:settings, @raise_on_failure) do
           @index.settings(*args)
         rescue ::MeiliSearch::ApiError => e
-          return {} if e.code == 404 # not fatal
+          return {} if e.code == 'index_not_found' # not fatal
```

Comment:
I'm not sure why we had the 404 in the first place, but `index_not_found` is just one issue that could happen, shouldn't we make it more generic? Like parsing the error response to grab a "match-all" case?

Reply:
I think the logic is that everytime an index is used, we attempt to create it __and__ we also check if its settings are good, immediately afterward.

So
- if `index.settings` says that the index is not found
- it means that the index has just been created
- thus we can pass `{}` as the settings to compare against the user settings
- this way if there are any user settings at all (we know the index has just been created) we will send an `update_settings` request

This logic has a flaw that we should discuss in due time but it's minor.

Anyway from what I can see the only purpose of this rescue block is to handle this one case wherein the index has not been created yet. Other errors are handled by `log_or_throw` since this method just propagates them.